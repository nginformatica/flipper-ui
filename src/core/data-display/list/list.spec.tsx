import * as React from 'react'
import { render, screen } from '@testing-library/react'
import ListItem from '@/core/data-display/list-item'
import { List } from '.'

describe('List', () => {
    it('should render', () => {
        render(
            <List>
                <ListItem>Item 1</ListItem>
                <ListItem>Item 2</ListItem>
                <ListItem>Item 3</ListItem>
            </List>
        )

        const list = screen.queryAllByRole('menuitem')

        expect(list).toHaveLength(3)
    })

    it('should render with title', () => {
        render(
            <List title='My beautiful list'>
                <ListItem>Item 1</ListItem>
                <ListItem>Item 2</ListItem>
                <ListItem>Item 3</ListItem>
            </List>
        )

        const list = screen.queryAllByRole('menuitem')
        const title = screen.queryByText('My beautiful list')

        expect(list).toHaveLength(3)
        expect(title).toBeDefined()
    })

    it('should render with custom className', () => {
        render(
            <List data-testid='list-container' className='my-custom-class'>
                <ListItem>Item 1</ListItem>
                <ListItem>Item 2</ListItem>
                <ListItem>Item 3</ListItem>
            </List>
        )

        const listContainer = screen.getByTestId('list-container')

        expect(listContainer.classList).toContain('my-custom-class')
    })
})
