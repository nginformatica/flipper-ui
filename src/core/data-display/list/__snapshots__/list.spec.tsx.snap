// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<div>
  <ul
    class="MuiList-root makeStyles-default-19 my-custom-class MuiList-padding"
    data-testid="list-container"
  >
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-root-23 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      Item 1
      <span
        class="MuiTouchRipple-root"
      />
    </li>
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-root-23 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      Item 2
      <span
        class="MuiTouchRipple-root"
      />
    </li>
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-root-23 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      Item 3
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </ul>
</div>
`;
